Unable to re-nalayse samples

**Environment information**
* MWDB version (from `/about`): [2.8.0](https://github.com/CERT-Polska/mwdb-core/releases/tag/v2.8.0)
* Installation method:
  - [ ] mwdb.cert.pl service
  - [x] From PyPi (`pip install mwdb-core`)
  - [ ] From docker-compose
  - [ ] Other (please explain)
* Plugins installed:

**Behaviour the bug (what happened?)**

Got an "Internal Server Error" when I'm clicking on reanalysed on a sample.

**Expected behaviour**

The samples to be reanalysed

**Reproduction Steps**

1. Go to a sample
2. click on "reanalyse"
3. Get a "Internal Server Error"

**Additional context**

```
Oct 16 13:48:32 stalkie uwsgi[1534]: [pid: 1534|app: 0|req: 187/187] 127.0.0.1 () {46 vars in 1321 bytes} [Sun Oct 16 13:48:32 2022] GET /api/object/b474cf96537fe8eb1d5d31c29ca096297c610aacd9fdb4913bb80ddcad223976/comment => generated 3 bytes in 76 msecs (HTTP/1.0 200) 2 headers in 70 bytes (1 switches on core 0)
Oct 16 13:48:34 stalkie sshd[1780]: Connection closed by 143.198.53.72 port 50352 [preauth]
Oct 16 13:48:41 stalkie uwsgi[1534]: [ERROR] uWSGIWorker1Core0 - service.error_router:76 - Unhandled exception occurred
Oct 16 13:48:41 stalkie uwsgi[1534]: Traceback (most recent call last):
Oct 16 13:48:41 stalkie uwsgi[1534]:   File "/home/mwdb/venv/lib/python3.9/site-packages/flask/app.py", line 1516, in full_dispatch_request
Oct 16 13:48:41 stalkie uwsgi[1534]:     rv = self.dispatch_request()
Oct 16 13:48:41 stalkie uwsgi[1534]:   File "/home/mwdb/venv/lib/python3.9/site-packages/flask/app.py", line 1502, in dispatch_request
Oct 16 13:48:41 stalkie uwsgi[1534]:     return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
Oct 16 13:48:41 stalkie uwsgi[1534]:   File "/home/mwdb/venv/lib/python3.9/site-packages/flask_restful/__init__.py", line 467, in wrapper
Oct 16 13:48:41 stalkie uwsgi[1534]:     resp = resource(*args, **kwargs)
Oct 16 13:48:41 stalkie uwsgi[1534]:   File "/home/mwdb/venv/lib/python3.9/site-packages/flask_limiter/extension.py", line 1036, in __inner
Oct 16 13:48:41 stalkie uwsgi[1534]:     return current_app.ensure_sync(obj)(*a, **k)
Oct 16 13:48:41 stalkie uwsgi[1534]:   File "/home/mwdb/venv/lib/python3.9/site-packages/flask_limiter/extension.py", line 1036, in __inner
Oct 16 13:48:41 stalkie uwsgi[1534]:     return current_app.ensure_sync(obj)(*a, **k)
Oct 16 13:48:41 stalkie uwsgi[1534]:   File "/home/mwdb/venv/lib/python3.9/site-packages/flask_limiter/extension.py", line 1036, in __inner
Oct 16 13:48:41 stalkie uwsgi[1534]:     return current_app.ensure_sync(obj)(*a, **k)
Oct 16 13:48:41 stalkie uwsgi[1534]:   [Previous line repeated 5 more times]
Oct 16 13:48:41 stalkie uwsgi[1534]:   File "/home/mwdb/venv/lib/python3.9/site-packages/flask/views.py", line 84, in view
Oct 16 13:48:41 stalkie uwsgi[1534]:     return current_app.ensure_sync(self.dispatch_request)(*args, **kwargs)
Oct 16 13:48:41 stalkie uwsgi[1534]:   File "/home/mwdb/venv/lib/python3.9/site-packages/flask_restful/__init__.py", line 582, in dispatch_request
Oct 16 13:48:41 stalkie uwsgi[1534]:     resp = meth(*args, **kwargs)
Oct 16 13:48:41 stalkie uwsgi[1534]:   File "/home/mwdb/venv/lib/python3.9/site-packages/mwdb/resources/__init__.py", line 47, in endpoint
Oct 16 13:48:41 stalkie uwsgi[1534]:     return f(*args, **kwargs)
Oct 16 13:48:41 stalkie uwsgi[1534]:   File "/home/mwdb/venv/lib/python3.9/site-packages/mwdb/resources/__init__.py", line 31, in endpoint
Oct 16 13:48:41 stalkie uwsgi[1534]:     return f(*args, **kwargs)
Oct 16 13:48:41 stalkie uwsgi[1534]:   File "/home/mwdb/venv/lib/python3.9/site-packages/mwdb/resources/karton.py", line 129, in post
Oct 16 13:48:41 stalkie uwsgi[1534]:     analysis = db_object.spawn_analysis(obj["arguments"])
Oct 16 13:48:41 stalkie uwsgi[1534]:   File "/home/mwdb/venv/lib/python3.9/site-packages/mwdb/model/object.py", line 921, in spawn_analysis
Oct 16 13:48:41 stalkie uwsgi[1534]:     analysis_id = self._send_to_karton()
Oct 16 13:48:41 stalkie uwsgi[1534]:   File "/home/mwdb/venv/lib/python3.9/site-packages/mwdb/model/file.py", line 322, in _send_to_karton
Oct 16 13:48:41 stalkie uwsgi[1534]:     return send_file_to_karton(self)
Oct 16 13:48:41 stalkie uwsgi[1534]:   File "/home/mwdb/venv/lib/python3.9/site-packages/mwdb/core/karton.py", line 39, in send_file_to_karton
Oct 16 13:48:41 stalkie uwsgi[1534]:     producer.send_task(task)
Oct 16 13:48:41 stalkie uwsgi[1534]:   File "/home/mwdb/venv/lib/python3.9/site-packages/karton/core/karton.py", line 87, in send_task
Oct 16 13:48:41 stalkie uwsgi[1534]:     self.backend.register_task(task)
Oct 16 13:48:41 stalkie uwsgi[1534]:   File "/home/mwdb/venv/lib/python3.9/site-packages/karton/core/backend.py", line 300, in register_task
Oct 16 13:48:41 stalkie uwsgi[1534]:     rs.set(f"{KARTON_TASK_NAMESPACE}:{task.uid}", task.serialize())
Oct 16 13:48:41 stalkie uwsgi[1534]:   File "/home/mwdb/venv/lib/python3.9/site-packages/karton/core/task.py", line 245, in serialize
Oct 16 13:48:41 stalkie uwsgi[1534]:     return json.dumps(
Oct 16 13:48:41 stalkie uwsgi[1534]:   File "/usr/lib/python3.9/json/__init__.py", line 234, in dumps
Oct 16 13:48:41 stalkie uwsgi[1534]:     return cls(
Oct 16 13:48:41 stalkie uwsgi[1534]:   File "/usr/lib/python3.9/json/encoder.py", line 199, in encode
Oct 16 13:48:41 stalkie uwsgi[1534]:     chunks = self.iterencode(o, _one_shot=True)
Oct 16 13:48:41 stalkie uwsgi[1534]:   File "/usr/lib/python3.9/json/encoder.py", line 257, in iterencode
Oct 16 13:48:41 stalkie uwsgi[1534]:     return _iterencode(o, 0)
Oct 16 13:48:41 stalkie uwsgi[1534]:   File "/home/mwdb/venv/lib/python3.9/site-packages/karton/core/task.py", line 242, in default
Oct 16 13:48:41 stalkie uwsgi[1534]:     return {"__karton_resource__": obj.to_dict()}
Oct 16 13:48:41 stalkie uwsgi[1534]:   File "/home/mwdb/venv/lib/python3.9/site-packages/karton/core/resource.py", line 135, in to_dict
Oct 16 13:48:41 stalkie uwsgi[1534]:     "size": self.size,
Oct 16 13:48:41 stalkie uwsgi[1534]:   File "/home/mwdb/venv/lib/python3.9/site-packages/karton/core/resource.py", line 223, in size
Oct 16 13:48:41 stalkie uwsgi[1534]:     self.fd.seek(0, os.SEEK_END)
Oct 16 13:48:41 stalkie uwsgi[1534]: io.UnsupportedOperation: seek
Oct 16 13:48:41 stalkie uwsgi[1534]: [INFO] uWSGIWorker1Core0 - app.log_request:146 - request - path:/api/object/b474cf96537fe8eb1d5d31c29ca096297c610aacd9fdb4913bb80ddcad223976/karton - arguments:{} - method:POST - status:500 - response_time:0:00:00.107700 - response_size:37
```

--- a/docker-compose-dev-karton.yml
+++ b/docker-compose-dev-karton.yml
@@ -4,11 +4,12 @@ version: "3.3"
 services:
   minio:
     image: minio/minio
-    command: server /data
+    command: "server --address 0.0.0.0:9000 --console-address :8070 /data"
     volumes:
       - /tmp/minio:/data
     ports:
       - "127.0.0.1:9000:9000"
+      - "127.0.0.1:8070:8070"
     environment:
       - MINIO_ROOT_USER=mwdb-test-access
       - MINIO_ROOT_PASSWORD=mwdb-test-key
@@ -34,12 +35,12 @@ services:
       MWDB_ENABLE_REGISTRATION: 1
       MWDB_ENABLE_KARTON: 1
       # Uncomment if you want to test S3 functions
-      # MWDB_STORAGE_PROVIDER: s3
-      # MWDB_HASH_PATHING: 0
-      # MWDB_S3_STORAGE_ENDPOINT: "minio:9000"
-      # MWDB_S3_STORAGE_ACCESS_KEY: "mwdb-test-access"
-      # MWDB_S3_STORAGE_SECRET_KEY: "mwdb-test-key"
-      # MWDB_S3_STORAGE_BUCKET_NAME: "mwdb"
+      MWDB_STORAGE_PROVIDER: s3
+      MWDB_HASH_PATHING: 0
+      MWDB_S3_STORAGE_ENDPOINT: "minio:9000"
+      MWDB_S3_STORAGE_ACCESS_KEY: "mwdb-test-access"
+      MWDB_S3_STORAGE_SECRET_KEY: "mwdb-test-key"
+      MWDB_S3_STORAGE_BUCKET_NAME: "mwdb"
     volumes:
       - "./docker/mail_templates:/app/mail_templates"
       - "./mwdb:/app/mwdb"

--- a/mwdb/model/file.py
+++ b/mwdb/model/file.py
@@ -240,8 +240,10 @@ class File(Object):
                     Fileobj=stream,
                 )
                 stream.seek(0, io.SEEK_SET)
-            finally:
+                return stream
+            except Exception:
                 stream.close()
+                raise
         elif app_config.mwdb.storage_provider == StorageProviderType.DISK:
             return open(self._calculate_path(), "rb")
         else:

--- a/mwdb/model/file.py
+++ b/mwdb/model/file.py
@@ -223,19 +223,25 @@ class File(Object):
                 stream.seek(0, os.SEEK_SET)
                 return stream
         if app_config.mwdb.storage_provider == StorageProviderType.S3:
-            return get_s3_client(
-                app_config.mwdb.s3_storage_endpoint,
-                app_config.mwdb.s3_storage_access_key,
-                app_config.mwdb.s3_storage_secret_key,
-                app_config.mwdb.s3_storage_region_name,
-                app_config.mwdb.s3_storage_secure,
-                app_config.mwdb.s3_storage_iam_auth,
-            ).get_object(
-                Bucket=app_config.mwdb.s3_storage_bucket_name,
-                Key=self._calculate_path(),
-            )[
-                "Body"
-            ]
+            # Stream coming from Boto3 get_object is not buffered and not seekable.
+            # We need to download it to the temporary file first.
+            stream = tempfile.TemporaryFile(mode="w+b")
+            try:
+                get_s3_client(
+                    app_config.mwdb.s3_storage_endpoint,
+                    app_config.mwdb.s3_storage_access_key,
+                    app_config.mwdb.s3_storage_secret_key,
+                    app_config.mwdb.s3_storage_region_name,
+                    app_config.mwdb.s3_storage_secure,
+                    app_config.mwdb.s3_storage_iam_auth,
+                ).download_fileobj(
+                    Bucket=app_config.mwdb.s3_storage_bucket_name,
+                    Key=self._calculate_path(),
+                    Fileobj=stream,
+                )
+                stream.seek(0, io.SEEK_SET)
+            finally:
+                stream.close()
         elif app_config.mwdb.storage_provider == StorageProviderType.DISK:
             return open(self._calculate_path(), "rb")
         else:

